example/weather/services/tester/run_tests.go

```diff
@@ -26,54 +28,76 @@ func endTest(tr *gentester.TestResult, start time.Time, tc *TestCollection, resu
 	tc.AppendTestResult(results...)
 }
 
-func getStackTrace(wg *sync.WaitGroup, m *sync.Mutex) string {
+func getStackTrace(wg *sync.WaitGroup, m *sync.Mutex) (string, error) {
 	m.Lock()
 	defer wg.Done()
 	defer m.Unlock()
 	// keep backup of the real stderr
 	old := os.Stderr
 	f, w, _ := os.Pipe()
 	os.Stderr = w
-	defer w.Close()
 
 	debug.PrintStack()
+	w.Close()
 
-	outC := make(chan string)
+	outC := make(chan string, 1)
+	outErr := make(chan error, 1)
 	go func() {
 		var buf bytes.Buffer
-		io.Copy(&buf, f) // nolint: errcheck
-		outC <- buf.String()
+		_, err := io.Copy(&buf, f)
+		if err != nil {
```

Comment:
I am not sure the `if` is required. If the error is nil then `outErr <- err` works fine and if there's an error then `buf.String()` should return `""` (buffer is empty).

Reply:
OK cool. I was being overly cautious. :)  Will change